src/main/client_context.cpp

```diff
@@ -504,16 +505,23 @@ std::unique_ptr<QueryResult> ClientContext::executeNoLock(PreparedStatement* pre
             resultFT =
                 localDatabase->queryProcessor->execute(physicalPlan.get(), executionContext.get());
             if (this->transactionContext->isAutoTransaction()) {
-                this->transactionContext->commit();
+                this->transactionContext->commit(true);
+                autoCommitTriggered = true;
             }
         }
     } catch (std::exception& e) {
         transactionContext->rollback();
-        getMemoryManager()->getBufferManager()->getSpillerOrSkip(
-            [](auto& spiller) { spiller.clearFile(); });
-        progressBar->endProgress(executionContext->queryID);
-        return queryResultWithError(e.what());
+        return handleFailedExecute(executionContext.get(), e);
+    }
+
+    if (autoCommitTriggered) {
```

Comment:
I wonder if you can throw a CheckpointException and catch here instead of explicitly separating autoCheckpoint and commit in different places.

Reply:
probably a good idea, that should clean up the changes quite a bit

Reply:
updated